### DISCUSSION TITLE
Return Printing after Powerloss issue..

### DISCUSSION THREAD

**Discussion by VashBaldeus** [Jyers/Marlin#808] [Q&A] [answered]
I had a 16 hour print running, and some time during the night there was a spike of electricity that was enough for my printer to lose all power, in the morning when I woke up, the return to print ui asking if you'd like to return to printing was not appearing. I highly suspect that it disappeared after a while due to the fact no one gave it an answer as nobody touched the printer before I woke up.

I did double check if the resume feature was on, and it is. I do know for a fact it does work as I've had prints in the past and even last week resume after power loss when given an "ok" to resume on the ui.

**Comment by Jyers** [comment]
Interesting, I'll take a look at the power loss feature to see if there is anything missing. You're printing from sd I assume

**Reply by VashBaldeus** [reply]
> 
> 
> Interesting, I'll take a look at the power loss feature to see if there is anything missing. You're printing from sd I assume

Yes, using an SD card.

**Comment by VashBaldeus** [comment]
> Interesting, I'll take a look at the power loss feature to see if there is anything missing. You're printing from sd I assume

Also similar thing happens with Filament Runout sensor, in settings it is on, but when it is supposed to be triggered it just keeps printing..

**Comment by marcribe** [comment]
I had the same problem this night.
Printing from SD

**Comment by VashBaldeus** [comment]
@Jyers happened to me today, caught it live, happened on 3rd/4th layer when power went out, printer turned on, no resume menu popped up. From my testing is is not due to the fact my *.gcode is inside a folder, since I've tested and resume worked both from root and a folder. this is very weird.

**Comment by Cal-LF** [comment]
Did a trial print with SD card for power loss. Activated and saved the Power Loss feature in the Advanced menu. No resume print screen appears. Restarts with the main menu.

**Reply by gothcha** [reply]
I concur, PLR file is being stored on SD but no pop-up on restart.

**Comment by Jyers** [comment]
This is an artifact of leaving ExtUI. I'm working on finding the best fix.

**Reply by VashBaldeus** [reply]
> 
> 
> This is an artifact of leaving ExtUI. I'm working on finding the best fix.

I am on 1.3.2, a version before leaving ExtUI..

**Reply by Jyers** [reply]
Interesting, well either way, I know what the issue is so a fix should come soon.

**Reply by Vignesh2659** [reply]
Thanks Jyers. Had the same issue and opened a new thread without seeing this. Shall wait for your 1.3.4 upgrade. Thanks

**Comment by Jyers** [comment, marked as answer]
This should be fixed now :)

**Reply by Cal-LF** [reply]
Tried the power loss return. I does offer to continue the print on power up. However It will not let the print stop after resuming. No action happens when clicking confirm to stop the print. Cancel brings back to the tune menu. Clicking stop again brings to the confirm. Turning printer off and on brings back to resume print.

![Image01](https://user-images.githubusercontent.com/76499230/120565365-dfcd6980-c3da-11eb-88f0-a14e07236241.jpg)
![Image02](https://user-images.githubusercontent.com/76499230/120565371-e52ab400-c3da-11eb-93d8-28506d591d61.jpg)
![Image03](https://user-images.githubusercontent.com/76499230/120565387-ea87fe80-c3da-11eb-9ab7-c5a4f1466fd9.jpg)
![Image04](https://user-images.githubusercontent.com/76499230/120565389-ed82ef00-c3da-11eb-955c-10b83fe9f1c0.jpg)

**Reply by Jyers** [reply]
I think I've got a fix for this, just need to test it real quick!

**Reply by Jyers** [reply]
@Cal-LF Give the newest commit a try :)

**Reply by Cal-LF** [reply]
The print resumes but starts completely over at the purge lines on the edge of the bed.

**Reply by Cal-LF** [reply]
The nozzle returned to the position on the previous try, just could not stop the print.

**Reply by Cal-LF** [reply]
@Jyers @tititopher68-dev Yes sir, but as I said above, now when it returns to print after power loss, it does not go back to where it left off and finish the print. Instead it starts over as a new print including the purge lines on the left of the bed. The stop and pause function works as you stated.

**Reply by Cal-LF** [reply]
I really have not used or needed the power loss feature. I turn it off as default when compiling my version. I believe it may cause early failure of the SD card by continually writing the PLR file while reading the gcode file.
I am seeing it is important to many users so have tried to help by testing the function.

**Reply by Jyers** [reply]
@Cal-LF are you sure it's doing the purge line and not just homing? I haven't changed anything about how the recovery is initiated so I don't see why it would be doing that.

**Reply by Cal-LF** [reply]
@Jyers Retested the power loss and it resumed print where it left off. Did not go to the purge line so may have been operator error. I did install the latest commit today, that is the only difference. 
I like the change made to the preheat menu. Zeroing out the temps with a new select is nice.

**Reply by Jyers** [reply]
Glad you like it! You have @gothcha to thank for that one :P
